chore: Enabled disabling of transactions

PhilWindle opened the pull request:
This PR adds a configuration option to disable transactions in the node. This means transactions received over RPC or P2P will be rejected. We also don't subscribe to tx gossip or tx retrieval requests.

alexghr commented on yarn-project/aztec-node/src/aztec-node/server.ts:
```diff
@@ -1082,6 +1082,7 @@ export class AztecNodeService implements AztecNode, AztecNodeAdmin, Traceable {
       setupAllowList: this.config.txPublicSetupAllowList ?? (await getDefaultAllowedSetupFunctions()),
       gasFees: await this.getCurrentBaseFees(),
       skipFeeEnforcement,
+      txsPermitted: !this.config.disableTransactions,
```
Hmm, this is fine but would have preferred to not have to negate the boolean.

critesjosh committed:
chore: Enabled disabling of transactions

This PR adds a configuration option to disable transactions in the node. This means transactions received over RPC or P2P will be rejected. We also don't subscribe to tx gossip or tx retrieval requests.
